### DISCUSSION TITLE
Force qutebrowser to run in XWayland

### DISCUSSION THREAD

**Discussion by joaonunatings** [qutebrowser/qutebrowser#7025] [Q&A] [answered]
Is there any way to force qutebrowser open using XWayland for communication purposes (redirect page automatically opens X application)?

**Comment by The-Compiler** [comment, marked as answer]
I don't follow. What do you mean with "for communication purposes"? Do you have an example?

**Reply by joaonunatings** [reply]
Yes, I did encounter a problem today. I was launching Unity Hub, it asked me to sign in. It opened qutebrowser, sucessfully signed in but no redirection back to the application. I tried using firefox running as X Client and worked with no problems. I assumed this would be wayland - xwayland communication issues. But maybe I'm wrong? Also in firefox, a pop-up appeared so I could 'Open with application'.

**Reply by toofar** [reply]
Probably you need to set the [content.unknown_url_scheme_policy](https://qutebrowser.org/doc/help/settings.html#content.unknown_url_scheme_policy) setting to tell qutebrowser it is allowed to open the external application. I don't think we support interactive prompts for this case, don't remember why.
That setting supports [URL Patterns](https://qutebrowser.org/doc/help/configuring.html#patterns) and you should absolutely use them in this case!
So it'll be something like `:set -u www.the-login-site.com content.unknown_url_scheme_policy allow-all` where `www.the-login-site.com` is the full hostname of the site you are currently stuck on on after signing in.

Additionally it's possible that qutebrowser finds what native application to hand off to differently than firefox. I think firefox can use gconf stuff where qutebrowser only uses .desktop files (via xdg-open). Probably Unity Hub ships with one of those though.

**Reply by joaonunatings** [reply]
@toofar Thank you very much. Setting [content.unknown_url_scheme_policy](https://qutebrowser.org/doc/help/settings.html#content.unknown_url_scheme_policy) to the Unity redirect page worked. In this case: `set -u *://api.unity.com/* unknown_url_scheme_policy allow-all`
